flink-batch-connectors/flink-avro/src/main/java/org/apache/flink/api/java/io/AvroInputFormat.java

```diff
@@ -119,12 +138,14 @@ public E nextRecord(E reuseValue) throws IOException {
 		if (reachedEnd()) {
 			return null;
 		}
-		
-		if (!reuseAvroValue) {
-			reuseValue = InstantiationUtil.instantiate(avroValueType, Object.class);
+		if (org.apache.avro.generic.GenericRecord.class == avroValueType) {
```

Comment:
I think we can save one comparison for the `reuseAvroValue == true` case if we move that check up:
```
if (reuseAvroValue) {
  return dataFileReader.next(reuseValue);
} else {
  if (GenericRecord.class == avroValueType) {
    return dataFileReader.next();
  } else {
    return dataFileReader.next(InstantiationUtil.instantiate(avroValueType, Object.class));
  }
}
```